ntoskrnl/mm/mmfault.c

```diff
@@ -248,11 +248,11 @@ MmAccessFault(IN BOOLEAN StoreInstruction,
     {
         /* This is an ARM3 fault */
         DPRINT("ARM3 fault %p\n", MemoryArea);
-        return MmArmAccessFault(StoreInstruction, Address, Mode, TrapInformation);
+        return MmArmAccessFault(FaultCode, Address, Mode, TrapInformation);
     }
 
     /* Keep same old ReactOS Behaviour */
-    if (StoreInstruction)
+    if (MI_IS_WRITE_ACCESS(FaultCode))
```

Comment:
This should be `!MI_IS_NOT_PRESENT_FAULT` unless you're planning on changing the semantics